assets/source/setup-guide/app/components/HealthCheck/index.js

```diff
@@ -0,0 +1,252 @@
+};
+
+const FormattedReasons = ( { reasons } ) => {
+	if ( reasons === undefined ) {
+		return null;
+	}
+
+	return (
+		<ul className="ul-square">
+			{ reasons.map( ( reason ) => (
+				<li key={ reason }>{ DISAPPROVAL_COPY_STATES[ reason ] }</li>
+			) ) }
+		</ul>
+	);
+};
+
+const HealthCheck = () => {
+	const createNotice = useCreateNotice();
+	const appSettings = useSettingsSelect();
+	const [ healthStatus, setHealthStatus ] = useState();
+
+	useEffect( () => {
+		checkHealth();
+	}, [ checkHealth ] );
+
+	const checkHealth = useCallback( async () => {
+		try {
+			const results = await apiFetch( {
+				path:
+					wcSettings.pinterest_for_woocommerce.apiRoute + '/health/',
+				method: 'GET',
+			} );
+
+			setHealthStatus( results );
+		} catch ( error ) {
+			createNotice(
+				'error',
+				error.message ||
+					__(
+						'Couldn’t retrieve the health status of your account.',
+						'pinterest-for-woocommerce'
+					)
+			);
+		}
+	}, [ createNotice ] );
+
+	if ( healthStatus === undefined || healthStatus.status === 'approved' ) {
+		return null;
+	}
+
+	const notices = {
+		pending_initial_configuration: {
+			status: 'warning',
+			message: __(
+				'The feed is being configured. Depending on the number of products this may take a while as the feed needs to be fully generated before its been sent to Pinterest for registration. You can check the the status of the generation process in the Catalog tab.',
+				'pinterest-for-woocommerce'
+			),
+			dismissible: false,
+		},
+		pending: {
+			status: 'warning',
+			message: __(
+				'Please hold on tight as your account is pending approval from Pinterest. This may take up to 5 business days.',
+				'pinterest-for-woocommerce'
+			),
+			dismissible: false,
+		},
+		declined: {
+			status: 'error',
+			message: __(
+				'Your merchant account is disapproved.',
+				'pinterest-for-woocommerce'
+			),
+			body: __(
+				'If you have a valid reason (such as having corrected the violations that resulted in the dissaproval) for appealing a merchant review decision, you can submit an appeal.',
+				'pinterest-for-woocommerce'
+			),
+			reasons: healthStatus.reasons,
+			dismissible: false,
+			actions: [
+				{
+					label: __(
+						'Submit an appeal',
+						'pinterest-for-woocommerce'
+					),
+					url: addQueryArgs(
+						wcSettings.pinterest_for_woocommerce.pinterestLinks
+							.appealDeclinedMerchant,
+						{ advertiserId: appSettings.tracking_advertiser }
+					),
+				},
+			],
+		},
+		appeal_pending: {
+			status: 'warning',
+			dismissible: false,
+			message: __(
+				'Your merchant account is disapproved.',
+				'pinterest-for-woocommerce'
+			),
+			body: __(
+				'Please hold on tight as there is an Appeal pending for your Pinterest account.',
+				'pinterest-for-woocommerce'
+			),
+		},
+		error: {
+			status: 'error',
+			dismissible: false,
+		},
+	};
+
+	const notice = notices[ healthStatus.status ] || notices.error;
+
+	if ( healthStatus.status === 'error' ) {
```

Comment:
If the intention here is to handle unknown errors by the fallback message "Could not fetch account status" below, it would need to adjust this condition. Otherwise, it would show a blank error notice block.

![image](https://user-images.githubusercontent.com/17420811/138053267-af2d0cf0-70e3-4ba7-829f-66f7146a13c1.png)


```suggestion
	if ( notice.status === 'error' ) {
```